repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeServlet.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.powerfilter
+
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import javax.inject.{Inject, Named}
+import javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR
+import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
+import org.openrepose.commons.utils.servlet.http.{HttpServletRequestWrapper, HttpServletResponseWrapper, ResponseMode}
+
+@Named
+class ReposeServlet @Inject()(router: PowerFilterRouter) extends HttpServlet with StrictLogging {
+
+  override def service(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
+    val wrappedRequest = new HttpServletRequestWrapper(req)
+    val wrappedResponse = new HttpServletResponseWrapper(resp, ResponseMode.MUTABLE, ResponseMode.PASSTHROUGH)
+
+    try {
+      if (wrappedResponse.getStatus() < SC_INTERNAL_SERVER_ERROR) {
```

Comment:
I wasn't actually sure whether or not we wanted to keep this check. I left it in since that's the current behavior, but honestly... I don't particularly like it.

Reply:
This just short circuits the request from going to the origin if it is a 5xx.
Although the filters that care about this already short circuit if not in delegation mode.
What were your thoughts/reasoning on not doing this?

Reply:
For one, as you pointed out, if a filter wants to reject a request, it will probably do so itself, and not rely on processing from subsequent filters or the filter chain. This conditional would allow for further processing from subsequent filters, which may have some use case that I can't think of, but so does delegation and delegating mode. The difference is, delegation provides more information. Also, it's likely safer, since filters normally don't check the response status code before writing their own, meaning if the first filter set a `500` response, but a later filter set a `400` response without checking the current code, this condition would still be satisfied and the request would be routed. 

Secondly, this conditional only covers `5xx` status codes. What about `4xx` codes? Why only circumvent routing and rewriting the response on `5xx`s? This emphasis how niche this code is.

Reply:
Yeah, i'm not a fan. If we make it to the servlet send the request on to the OS. We shouldn't be half writing responses and hoping somebody else cleans up the mess. Delegating mode doesn't even do that, it just adds a request header saying here's what i would have done, but you told me not to.

Reply:
Sure, delegating mode is a little different, but it seems like both are meant to say, "I would have failed, but I'm putting that failure off to allow for further processing."

Anyway, great! I'll remove this. That makes me happy.

Reply:
Yeah, my policy in the filter chain and with all the other changes we are making to these bits, is if it seems like it might be stupid raise the flag, so that we can discuss and probably remove it.